pandas/core/internals/blocks.py

```diff
@@ -571,6 +571,43 @@ def copy(self, deep: bool = True) -> Self:
             refs = self.refs
         return type(self)(values, placement=self._mgr_locs, ndim=self.ndim, refs=refs)
 
+    # ---------------------------------------------------------------------
+    # Copy-on-Write Helpers
+
+    @final
+    def _maybe_copy(self, using_cow: bool, inplace: bool) -> Self:
+        if using_cow and inplace:
+            deep = self.refs.has_reference()
+            blk = self.copy(deep=deep)
```

Comment:
Just a general note here (it's not a problem right now): this could have side effects, creating a shallow copy of the block sets up another reference, which will yield True for ``blk.refs.has_reference()`` and ``self.refs.has_reference()``, so have to be a bit careful when using